ClinGen/clincoded
# Connects to #1711. Connects to #1712. Connects to #1739. Connects to #1763. Connects to #1773. Updates to classification publication feature (including updated segregation scoring).

Steps to test #1711:
1. Within a gene-disease record, click on the button to add case-control evidence.
2. Scroll to the bottom of the page to confirm that the "Score:" field is disabled (and the informational text is present).
3. Scroll up to the "Study type:" field and select an option.
4. Return to the "Score:" field to confirm that it’s now enabled (and the text is no longer present).


Steps to test #1712:
To confirm that the kafka client installation (via cloud-config.yml) was successful:
1. On an AWS instance, after manually adding the "KAFKA_CERT_PW" environment variable, run the following to check that the kafka client (v0.11.4) was successfully installed:
`sudo pip3 show confluent-kafka`

To confirm the column header change on the dashboard:
1. Start a new variant interpretation or add yourself to an affiliation with existing interpretations and then visit the home/dashboard page.

To confirm proper use of the new "classificationDate" property and updates to published data, approve and publish a classification:
1. As you’re going through the process, keep an eye on the timestamp for the "Last Saved Summary Classification".  It should only change when a classification is saved (not when other updates happen, like provisioning/approving the classification).
2. When approving the classification, be sure to add an approved date.  Once the classification has been published, this date should appear as the "EXPERT CURATION (DATE)" on the website.  Also, if "CALCULATED CLASSIFICATION (DATE)" and/or "MODIFIED CLASSIFICATION (DATE)" appear on the website, their value should match the "Last Saved Summary Classification" date from the GCI.
3. After the classification has been published, view the corresponding approved evidence summary.  It should include information regarding the classification’s publish status (label and "Date classification published" data).


Steps to test #1739:
1. If not already present in a published classification, add one or more families with segregation scoring and republish (going through the entire approval process).  When adding the scoring, select "Yes" for the "Published LOD score?" field and confirm that you can’t add a score of zero (in "Published Calculated LOD score").
2. Check the website to confirm that the published classification has the updated segregation scoring (split into "Candidate gene sequencing" and "Exome/genome or all genes sequenced in linkage region" categories), as well as evidence counts (in the "Count" column).


Steps to test #1763:
1. If not already present in a published classification, add one or more scored probands to families and republish (going through the entire approval process).
2. Check the website to confirm that the list of PMIDs associated with the segregation scoring is complete (from the GCI evidence summary, includes all articles with a counted LOD score from both the "Genetic Evidence: Case Level (family segregation information without proband data or scored proband data)" and "Genetic Evidence: Case Level (variants, segregation)" tables).
